File: src/main/java/cz/grunwjir/csasdemo/csas/client/CsasClientConfig.java
package cz.grunwjir.csasdemo.csas.client;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * Configuration class for Csas API client.
 *
 * @author Jiri Grunwald
 */
@Configuration
@EnableConfigurationProperties(CsasClientProperties.class)
public class CsasClientConfig {

    @Bean
    public CsasClient csasClient(CsasClientProperties clientProperties) {
        return new CsasClientImpl(clientProperties);
    }
}

File: src/main/kotlin/cz/grunwjir/csasdemo/csas/client/CsasClientConfig.kt
package cz.grunwjir.csasdemo.csas.client

import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * Configuration class for Csas API client.
 *
 * @author Jiri Grunwald
 */
@Configuration
@EnableConfigurationProperties(CsasClientProperties::class)
class CsasClientConfig {

    @Bean
    fun csasClient(clientProperties: CsasClientProperties): CsasClient = CsasClientImpl(clientProperties)
}